src/HandlesUpdatingPlugins.php

```diff
@@ -345,46 +364,29 @@ protected function buildUpdateInfoByVersion($versionInfo)
     }
 
     /**
-     * Sets local plugins update cache.
-     *
-     * By first unhooking the "checkIfUpdateIsAvailable" and then rehooking it, so that we won't be hitting the remote
-     * API along the way.
-     *
-     * @param object|null $updateCache
-     *
-     * @return void
-     */
-    protected function cachePluginUpdateInfo($updateCache)
-    {
-        remove_filter('pre_set_site_transient_update_plugins', $updateHook = [$this, 'checkIfUpdateIsAvailable']);
-
-        set_site_transient('update_plugins', $updateCache);
-
-        add_filter('pre_set_site_transient_update_plugins', $updateHook);
-    }
-
-    /**
      * Returns version info either from the cache or from the remote API.
      *
      * @return object|null
      */
     protected function getVersionInfo()
     {
-        if ($versionInfo = $this->getCachedVersionInfo()) {
+        if ($versionInfo = $this->getVersionInfoFromCache()) {
             return $versionInfo;
         }
 
-        return $this->getRemoteVersionInfo();
+        return $this->getVersionInfoFromApi();
     }
 
     /**
      * Returns version info from the local cache.
      *
-     * @return object|null
+     * @param string|null $cacheKey
+     *
+     * @return null|object
      */
-    protected function getCachedVersionInfo($cacheKey = '')
+    protected function getVersionInfoFromCache($cacheKey = null)
     {
-        empty($cacheKey) and $cacheKey = $this->getCacheKey();
+        $cacheKey === null and $cacheKey = $this->getCacheKey();
```

Comment:
I hadn't noticed this use of `and` before. Is the purpose here to stop executing this line of the first boolean check fails? i.e. if `$cacheKey !== null` then we don't need to call `$this->getCacheKey()`?.

Reply:
Exactly. It's more of a one-liner `if` statement. I find it more readable and terse. 

I'd rewrite it to `$cacheKey or $cacheKey = $this->getCacheKey` if the code related to the cache wasn't removed.